src/config.py

```diff
@@ -0,0 +1,387 @@
+
+        return self.variablesDef
+
+    @property
+    def routes(self):
+        """Return the routes definition.
+
+        :return: array
+            The routes array, in PHP nested array form.
+        :exception RuntimeException:
+            If the routes are not accessible due to being in the wrong environment.
+        """
+        if not self.is_valid_platform():
+            raise RuntimeError('You are not running on Platform.sh, so routes are not available.')
+
+        if self.in_build():
+            raise RuntimeError('Routes are not available during the build phase.')
+
+        return self.routesDef
+
+    def get_route(self, route_id):
+        """Get route definition by route ID.
+
+        :param route_id: string
+            The ID of the route to load.
+        :return: array
+            The route definition. The generated URL of the route is added as a 'url' key.
+        :exception ValueError:
+            If there is no route by that ID, an exception is thrown.
+
+        """
+
+        for (url, route) in self.routes.items():
+
+            if route['id'] == route_id:
+
+                route['url'] = url
+
+                return route
+
+        raise ValueError('No such route id found: {}'.format(route_id))
+
+    def application(self):
+        """Returns the application definition array.
+
+        This is, approximately, the .platform.app.yaml file as a nested array. However, it also has other information
+        added by Platform.sh as part of the build and deploy process.
+
+        :return: array
+            The application definition array.
+        """
+
+        if not self.is_valid_platform():
+
+            RuntimeError('You are not running on Platform.sh, so the application definition are not available.')
+
+        return self.applicationDef
+
+    def on_enterprise(self):
+        """Determines if the current environment is a Platform.sh Enterprise environment.
+
+        :return: bool
+            True on an Enterprise environment, False otherwise.
+        """
+
+        return self.is_valid_platform() and self.get_value('MODE') == 'enterprise'
+
+    def on_production(self):
+        """Determines if the current environment is a production environment.
+
+        Note: There may be a few edge cases where this is not entirely correct on Enterprise, if the production
+        branch is not named `production`. In that case you'll need to use your own logic.
+
+        :return: bool
+            True if the environment is a production environment, False otherwise.
+            It will also return False if not running on Platform.sh or in the build phase.
+
+        """
+
+        if not self.is_valid_platform() and not self.in_build():
+
+            return False
+
+        prod_branch = 'production' if self.on_enterprise() else 'master'
+
+        return self.get_value('BRANCH') == prod_branch
+
+    def get_value(self, name):
```

Comment:
Python has no concept of syntactic private methods.  Is there a convention we should use instead to indicate that outsiders are not supposed to use this method?  @NickAnderegg?

Reply:
From what I can tell the convention is to name it as `__get_value` to indicate that.

Reply:
@Crell Single underscore is basically protected, double underscore mangles the method, so it's basically private.

So `_somemethod` is accessed via `yourinstance._somemethod()`, while `__somemethod` is accessed via `yourinstance._ClassName__somemethod()`

Reply:
Double underscore should really only be used to avoid name collisions, so single underscore is preferred

Reply:
got it